pool/pool.go

```diff
@@ -1406,6 +1406,48 @@ const (
 	defaultStreamTimeout      = 10 * time.Second
 )
 
+// DefaultOptions returns default option preset for Pool creation. It may be used like start point for configuration or
+// like main configuration.
+func DefaultOptions(signer neofscrypto.Signer) InitParameters {
+	params := InitParameters{
+		signer:                    signer,
+		sessionExpirationDuration: defaultSessionTokenExpirationDuration,
+		errorThreshold:            defaultErrorThreshold,
+		clientRebalanceInterval:   defaultRebalanceInterval,
+		healthcheckTimeout:        defaultHealthcheckTimeout,
+		nodeDialTimeout:           defaultDialTimeout,
+		nodeStreamTimeout:         defaultStreamTimeout,
+	}
+
+	return params
+}
+
+// New creates connection pool using simple set of endpoints and parameters.
+//
+// See also [pool.DefaultOptions] for details.
+//
+// Returned errors:
+//   - [neofscrypto.ErrIncorrectSigner]
+func New(endpoints []string, options InitParameters) (*Pool, error) {
```

Comment:
How about
`New(endpoints []NodeParam, signer neofscrypto.Signer, options InitParameters) (*Pool, error)`
and
`NewFlatNodeParams(endpoints []string) []NodeParam`

This way we can replace `NewPool` completely. It's a little less convenient, but more versatile at the same time.

Reply:
Looks not so easy, but makes sense. Updated

Reply:
We can make them two methods, opinions are welcome, but I think one is enough.